python/core/qgsexpression.sip

```diff
@@ -357,8 +368,8 @@ class QgsExpression
         QgsExpression::Node* operand() const;
 
         virtual QgsExpression::NodeType nodeType() const;
-        virtual bool prepare( QgsExpression* parent, const QgsFields &fields );
-        virtual QVariant eval( QgsExpression* parent, const QgsFeature* f );
+        virtual bool prepare( QgsExpression* parent, const QgsExpressionContext& context );
+	virtual QVariant eval( QgsExpression* parent, const QgsExpressionContext* context );
         virtual QString dump() const;
```

Comment:
@mhugo good pickup. It wasn't by design, it was just because that was how the original methods were defined. I think I'll make them both take a const pointer, and if it's null, then evaluate/prepare the expression against the current project context by default.